Fixing tarball and testing page

ChrisChinchilla opened the pull request:
@jodok and @joemoe Let me know if you're happy with this. The page uses the old method of setting versions (config.json) to load testing and stable releases.

It fixes this - https://trello.com/c/fv35aTkQ/122-website-tarball-link-is-missing-whole-site-is-a-template

Not sure if it fixes these:

https://trello.com/c/NEWq2JNz/123-website-usability-to-find-link-to-tarball-at-the-new-website-is-really-worse
https://trello.com/c/iVTEyGRl/124-website-no-way-to-figure-out-latest-crate-versions-on-new-website

As it's still extra steps to get to these pages, but they are there…

ChrisChinchilla commented:
Not sure where that Ubuntu commit came from :/

jodok commented on _templates/base.html:
```diff
@@ -75,7 +75,7 @@
     <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/js/bootstrap.min.js" integrity="sha256-Sk3nkD6mLTMOF0EOpNtsIry+s1CsaqQC1rVLTAy+0yc= sha512-K1qjQ+NcF2TYO/eI3M6v8EiNYZfA95pQumfvcVrTHtwQVDG+aHRqLi/ETn2uB+1JqwYqVG3LIvdm9lj6imS/pQ==" crossorigin="anonymous"></script>    {% block extra_js %}
     {% endblock %}
-    <script async src="{% static '/static/js/main.js' %}"></script>
+    <script async src="{% static '/static/js/dl.js' %}"></script>
```
just add the tracking call inline. no need to include an additional javascript.
basic tracking is already included. all you need to do is to add 
{% block end_body %}
    <script>
      analytics.track('visit_download');
    </script>
{% endblock %}

ChrisChinchilla replied:
@jodok Yeah, I did that initially and it wouldn't work, but I'll try again and see…

ChrisChinchilla replied:
OK @jodok firstly this means constructing the button slightly differently as the previous code used JS to redirect to the download link, that's fine. But using the code above results in "Uncaught ReferenceError: analytics is not defined"

I'm just using an onclick event on a link at the moment, but to be honest I'm not really sure how we're tracking things anymore. If the old approach is no longer correct, it may be easier for you to finish the JS side of this :)

jodok replied:
you need to do this in the end_body section. analytics is defined later.

jodok replied:
i'd be happy to add the tracking later and show you the diff, if this is causing problems

jodok commented on static/js/dl.js:
```diff
@@ -0,0 +1,42 @@
+    }
+  });
+
+}(jQuery));
```
as mentioned, use segment tracking inline

joemoe commented:
with jodok's changes it is fine for me.

ChrisChinchilla commented:
@jodok and @joemoe I think this is done. Moved script to footer, changed analytics call, it seemed to work, I can't confirm the analytics though as I have no access.

jodok commented:
tracking is still not removed. please remove it. also the external js.

ChrisChinchilla commented:
I'm unclear @jodok it is removed and the js has been removed, both in the last commit. Unless I am misunderstanding what you are saying.

jodok commented:
if i look at the diff, i see them still there: https://github.com/crate/crate-web/pull/284/files

ChrisChinchilla commented:
@jodok I think you're looking at a template file that we're not actually using, there are a couple of template folders in the website and I wasn't sure which one we were using.

I will remove it from that too, and a commented out line with the old analytics code (just in case) but it was already removed from the template file we're actually using.

Apart from that, everything in the current checkout is using an inline script and the function you asked to use.

jodok commented on templates/wf_footer.html:
```diff
@@ -57,5 +57,26 @@
     analytics.page();
   }}();
 </script>
+{% block end_body %}
```
please remove this script as well. as it would be included in all docs pages, not only the download page. just skip tracking for now.

ChrisChinchilla commented:
OK @jodok this should be plain and simple for now. Would be nice to style the link, but at least it works for now :)

jodok commented:
cool :) seems like master is ahead of that branch and you need to resolve conflicts first.

ChrisChinchilla commented:
@jodok Yup, I have a lot of conflict resolving to do on several PRs now, Git fun beckons!
